syntax = "proto3";

service Booking {
    rpc GetBookingsByUserId(UserID) returns (BookingResponse) {}
    rpc GetBookings(EmptyBooking) returns (stream BookingResponse) {}
    rpc AddBookings(AddBooking) returns (BookingResponse) {}
}

message DateMovies {
    string date = 1;
    repeated string movies = 2;
}

message UserID {
    string userid = 1;
}

message AddBooking {
    string userid = 1;
    string date = 2;
    string movies = 3;
}

message BookingResponse {
    string userid = 1;
    repeated DateMovies datemovies = 2;
}

message EmptyBooking {
}